featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -736,3 +734,9 @@ def match(input_types, features, replace=False, commutative=False):
         return set([tuple(sorted(s, key=lambda x: x.get_name().lower())) for s in matching_inputs])
 
     return set([tuple(s) for s in matching_inputs])
+
+
+def ready_primitive(primitive):
+    if type(primitive) == type:
```

Comment:
this is how we do this check in FeatureBase

```
# initialize if not already initialized
if not isinstance(primitive, PrimitiveBase):
    primitive = primitive()
```

not sure which i like more. do you have thoughts on which way is better?